processor/executor/src/main/java/com/redhat/service/bridge/processor/actions/kafkatopic/KafkaTopicActionInvokerBuilder.java

```diff
@@ -12,22 +12,16 @@
 import org.eclipse.microprofile.reactive.messaging.Channel;
 import org.eclipse.microprofile.reactive.messaging.Emitter;
 
-import com.redhat.service.bridge.actions.ActionInvoker;
-import com.redhat.service.bridge.actions.ActionParameterValidator;
-import com.redhat.service.bridge.actions.InvokableActionProvider;
 import com.redhat.service.bridge.infra.exceptions.definitions.user.ActionProviderException;
 import com.redhat.service.bridge.infra.models.actions.BaseAction;
 import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+import com.redhat.service.bridge.processor.actions.ActionInvoker;
+import com.redhat.service.bridge.processor.actions.ActionInvokerBuilder;
 
 @ApplicationScoped
-public class KafkaTopicAction implements InvokableActionProvider {
-
-    public static final String TYPE = "KafkaTopic";
-
-    public static final String TOPIC_PARAM = "topic";
+public class KafkaTopicActionInvokerBuilder implements KafkaTopicActionBean, ActionInvokerBuilder {
```

Comment:
I did not get why we need the `KafkaTopicActionBean` and `KafkaTopicAction` (same for all the actions). 

iiuc, what you are trying to do with this abstraction can be achieved with generics removing the `..Bean` class for all the actions, for example you might turn this Invoker into something like 

```
public class KafkaTopicActionInvokerBuilder implements ActionInvokerBuilder<KafkaTopicAction> {
```

and change some classes accordingly.

Reply:
I agree that the split between `...Action` and `...ActionBean` is redundant and I just refactored it to have only interfaces.

However, the generics-based solution you are proposing is not enough because we don't directly inject the beans but, instead, we have an `ActionConfigurator` "factory" where they are retrieved by type. Thus, we need the `getType` and `accept` methods defined in `ActionBean` and the action-specific interface is explicitly used to set the correct type in all the beans belonging to the same action.